Sofa/Component/Collision/Response/Contact/src/sofa/component/collision/response/contact/AugmentedLagrangianResponse.inl

```diff
@@ -0,0 +1,65 @@
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <sofa/component/collision/response/contact/AugmentedLagrangianResponse.h>
+#include <sofa/core/visual/VisualParams.h>
+#include <sofa/component/collision/response/contact/CollisionResponse.h>
+#include <sofa/component/collision/response/mapper/BarycentricContactMapper.h>
+#include <sofa/component/collision/response/mapper/IdentityContactMapper.h>
+#include <sofa/component/collision/response/mapper/RigidContactMapper.inl>
+#include <sofa/simulation/Node.h>
+#include <iostream>
+
+namespace sofa::component::collision::response::contact
+{
+
+template < class TCollisionModel1, class TCollisionModel2, class ResponseDataTypes  >
+AugmentedLagrangianResponse<TCollisionModel1,TCollisionModel2,ResponseDataTypes>::AugmentedLagrangianResponse()
+    : AugmentedLagrangianResponse(nullptr, nullptr, nullptr)
+{
+}
+
+
+template < class TCollisionModel1, class TCollisionModel2, class ResponseDataTypes  >
+AugmentedLagrangianResponse<TCollisionModel1,TCollisionModel2,ResponseDataTypes>::AugmentedLagrangianResponse(CollisionModel1* model1, CollisionModel2* model2, Intersection* intersectionMethod)
+    : BaseUnilateralContactResponse<TCollisionModel1, TCollisionModel2, constraint::lagrangian::model::AugmentedLagrangianContactParameters, ResponseDataTypes>(model1,model2,intersectionMethod)
+    , d_mu (initData(&d_mu, 0.0, "mu", "friction coefficient (0 for frictionless contacts)"))
+    , d_epsilon (initData(&d_epsilon, 0.0, "epsilon", "Penalty parameter. It can be think of as a proportional controller, the lagrange multiplier is augmented by the violation multiplied by this factor at each solving iteration."))
+{
+
```

Comment:
Those tests are only to check if there is friction. I don't think that relying on a boolean member that need to be kept updated is a better mechanism. It makes code more complex and  error prone instead of testing it directly when needed. It is a simple double comparison, and it is not done that often that the computation difference makes it worth the potential errors.